internal/datastore/common/cursor.go

```diff
@@ -0,0 +1,49 @@
+package common
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	"github.com/authzed/spicedb/pkg/datastore"
+	"github.com/authzed/spicedb/pkg/datastore/options"
+	core "github.com/authzed/spicedb/pkg/proto/core/v1"
+)
+
+const (
+	errEncodingCursor = "error encoding cursor: %s"
+	errDecodingCursor = "error decoding cursor: %s"
+)
+
+func NoopCursorFunc(_ *core.RelationTuple) (options.Cursor, error) {
+	return "", datastore.ErrCursorsWithoutSorting
+}
+
+func EncodedProtoCursor(tpl *core.RelationTuple) (options.Cursor, error) {
```

Comment:
Is the cursor for resuming *at* the tuple or *after*? I assume after, but if so, can you clarify the parameter name here?

Reply:
After, but this isn't the right place to make that determination. The input represents the last tuple returned to the user, so I renamed it `last`.